yarpl/include/yarpl/flowable/Flowable.h

```diff
@@ -20,6 +20,21 @@
 namespace yarpl {
 namespace flowable {
 
+template <typename T>
+class Flowable;
+
+namespace detail {
+
+template <typename T>
+struct is_t_flowable : std::false_type {};
+
+template <typename R>
+struct is_t_flowable<Reference<Flowable<R>>> : std::true_type {
+  using ElemType = R;
```

Comment:
Nit: Using underscore-lowercase for `is_t_flowable` and then upper-camelcase for `ElemType` is a little odd.

I'd suggest picking one, either `is_t_flowable::value_type` or `IsFlowable::ElemType`.